### ISSUE TITLE
Request specific CI checks to be accepted

### ISSUE THREAD

**Issue by zoispag** [bobvanderlinden/probot-auto-merge#98]
Hi and thanks for the amazing bot.

I have a question.
We are using TeamCity for our builds, so it is possible that not all CI checks are already triggered on a PR.
Depending on the availability of our build agents, it is possible that a a build has finished and the others are not even queued yet, so the auto-merge merges it too soon.

Would it be possible to specify (by name) which checks should be satisfied?

**Comment by bobvanderlinden** [comment]
Yes, the first work was in #84, where auto-merge would fetch the required checks from the protected branch. This was however [later reverted](https://github.com/bobvanderlinden/probot-auto-merge/commit/8c3191338d9831f471ecb86f20558beb35e41766), because it didn't match GitHubs algorithm to determine whether checks are required. Documentation on this part was a bit flaky.

Currently I'm trying to move towards using GitHubs new GraphQL API and possibly using their preview APIs. One of the preview APIs includes `mergeStateStatus` (see [the fields of the PullRequest object](https://developer.github.com/v4/object/pullrequest/#fields)), which gives more information about the merge status than was possible with the old API. This will probably avoid doing my custom checks for protected branches and thus auto-merge can hold off merging based just on that field.

**Comment by zoispag** [comment]
Thanks for your reply @bobvanderlinden. For now I changed the **Branch protection rules** to require specific status checks and it works for me, but if it is configured through auto-merge config, it would be much better! :)
